### ISSUE TITLE
Services crash after /NS REGISTER

### ISSUE THREAD

**Issue by fuadragon** [anope/anope#368]
**Description**

Services crashed few times after i used /ns register passwd email

**Steps to reproduce the issue:**

1.
2.
3.

**Describe the results you received:**

https://paste.mozilla.org/Wih7f2JE

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `services --version`:**

Anope 2.1.2

**Comment by TehPeGaSuS** [comment]
I can't reproduce the issue on my network.

Services version: `Anope-2.1.3-git (gb26f198) -- build #2, compiled 21:08:32 Feb 25 2024, flags DG`

Services log of a user registering and then dropping the nickname (no confirmation, fake email):
```
<+NickServ> COMMAND: TheDog!~Wockets@2001:2002:2003:2004:2005:2006:2007:2008 used REGISTER to register TheDog (email: thedog@fake.org)
<+Global> Successfully delivered mail for TheDog (thedog@fake.org)
<+NickServ> COMMAND: TheDog!~Wockets@2a02:8429:629b:b701:6d17:5070:8f22:9e76 (TheDog) used DROP to drop nickname TheDog (group: TheDog) (email: thedog@fake.org)
<+NickServ> Deleting nickname group TheDog
```

**Comment by SadieCat** [comment]
This is already fixed in 2.1 git.
